### PR TITLE
Add the sample usecase and repository with GORM and WIRE

### PR DESCRIPTION
**Pull request by lovung**

### CODE REVIEW

File: pkg/gormutil/gormutil.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"gorm.io/gorm"
+)
+
+var (
+	once      sync.Once
+	singleton *gorm.DB
+)
+
+// OpenDBConnection opens a DB connection.
+func OpenDBConnection(connStr string, config gorm.Config) (*gorm.DB, error) {
```
Comment (phamh79) [inline pkg/gormutil/gormutil.go:16]: should we add mysql to the method name ? what do you think ?

**Comment by phamh79** [comment]
about the structure, how do you think if we have another service that use same entities, but may have different stuffs in handler/dto/config ?
for example: 
- a service for users which is exposed for public use
- a service for admin which is not exposed.

File: pkg/gormutil/gormutil.go (hunk 1 above)
Comment (lovung) [reply pkg/gormutil/gormutil.go:16]: You're right!

**Comment by lovung** [comment]
> about the structure, how do you think if we have another service that use same entities, but may have different stuffs in handler/dto/config ?
> for example:
> 
> * a service for users which is exposed for public use
> * a service for admin which is not exposed.

It's ok with the current structure. Just separate in the handler layer. Call the same use case.

**Pull request merged**